src/adagrams.js

```diff
@@ -1,15 +1,104 @@
+export const createLetterPool = () => {
+  const LETTERPOOL = {
+      "A":9,"B":2,"C":2,"D":4,
+      "E":12,"F":2,"G":3,"H":2,
+      "I":9,"J":1,"K":1,"L":4,"M":2,
+      "N":6,"O":8,"P":2,"Q":1,
+      "R":6,"S":4,"T":6,"U":4,
+      "V":2,"W":2,"X":1,
+      "Y":2,"Z":1
+  };
+
+  let letterList = [];
+  for (const [key,value] of Object.entries(LETTERPOOL)) {
+    for (let i = 0; i < value; i++) {
+      letterList.push(key);
+    };
+  };
+  return letterList;
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let letterPool = createLetterPool();
+
+  let letterHand = [];
+  for (let i =0; i < 10; i++) {
+    let randomIndex = Math.floor(Math.random() * letterPool.length);
+    let randomLetter = letterPool[randomIndex];
+    letterHand.push(randomLetter);
+    letterPool.splice(randomIndex,1);
```

Comment:
We usually want to avoid using splice inside a loop because it has O(n) time complexity. Each call can shift all remaining elements in the array, which adds up quickly in large datasets. This makes the loop less efficient, especially when performance matters. Instead, it's often better to use a different data structure like an object (or a Set or Map) for constant time lookups and deletions (O(1)). These structures provide more efficient ways to track and manage data when you need to repeatedly check or remove items during iteration.